pkg/test/client.go

```diff
@@ -16,13 +16,14 @@ import (
 )
 
 // NewFakeClient creates a fake K8s client with ability to override specific Get/List/Create/Update/StatusUpdate/Delete functions
-func NewFakeClient(t T, initObjs ...runtime.Object) *FakeClient {
+func NewFakeClient(t T, initObjs ...client.Object) *FakeClient {
 	s := scheme.Scheme
 	err := toolchainv1alpha1.AddToScheme(s)
 	require.NoError(t, err)
 	cl := fake.NewClientBuilder().
 		WithScheme(s).
-		WithRuntimeObjects(initObjs...).
+		WithObjects(initObjs...).
+		WithStatusSubresource(initObjs...).
```

Comment:
Related to our discussion in the host-operator PR, we could try to improve this logic to register status sub-resource for all "toolchain" resources. In this way we would know that, the status sub-resource is registered for all objects we manage. 
So, could you go through all toolchain GVKs and add them to the client builder as objects with the status sub-resource? Try to use a generic logic using the scheme and GVKs, we don't want to update the logic as soon as we add or remove  a CRD.

Reply:
added in [6dfea01](https://github.com/codeready-toolchain/toolchain-common/pull/376/commits/6dfea0122719082c1bfd61eafd6361989534c551)